### PR TITLE
fix devBuild

### PR DESCRIPTION
**Pull request by chenqingspring**
This change is [Reviewable](https://reviewable.io/reviews/shakacode/react_on_rails/877)

### CODE REVIEW

**Comment by justin808** [comment]
@mjhenkes This looks great. I'll examine this tomorrow. What are the reproduction steps to see the issue?

@Conturbo can you try this out?

**Comment by mjhenkes** [comment]
@justin808 Assuming a webpack config that looks something like [this](https://github.com/shakacode/react_on_rails/blob/master/lib/generators/react_on_rails/templates/base/base/client/webpack.config.js).

Then build assets for production (NODE_ENV=production webpack --config webpack.config.js) and the console will log "Webpack dev build for Rails". 
devBuild is always set to true because env (the list of environment variables) will never equal 'production'

@chenqingspring I'd recommend checking `const devBuild = configEnv !== 'production';` to stay inline the the fallbacks specified in the code.

**Comment by chenqingspring** [comment]
@mjhenkes It makes more sense :)

**Comment by justin808** [comment]
@chenqingspring Please add a changelog.md entry and I'll merge this. 

---

Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/877#-:-KntbcOTLB4nwc0PVwvr:b2bqasv)*

**Comment by justin808** [comment]
@chenqingspring 

1. CI failed linting
2. Please format the message for 8.0.4 as the release number **EXACTLY** like the other entries. And be sure to update both the top and the bottom of the file.

```
Finished in 1.38 seconds (files took 2.25 seconds to load)
1 example, 0 failures
Completed all RSpec tests
cd /home/travis/build/shakacode/react_on_rails && yarn run eslint
yarn run v0.24.6
$ eslint . 
/home/travis/build/shakacode/react_on_rails/webpackConfigLoader.js
  58:1  error  Trailing spaces not allowed  no-trailing-spaces
✖ 1 problem (1 error, 0 warnings)
```

**Comment by justin808** [comment]
Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[CHANGELOG.md, line 12 at r3](https://reviewable.io:443/reviews/shakacode/react_on_rails/877#-KnvriWg3U51gDNciIHU:-KnvriWg3U51gDNciIHV:blsve7x) ([raw file](https://github.com/shakacode/react_on_rails/blob/607a039ec836365421ebba9c5dd5874a86ea2562/CHANGELOG.md#L12)):*
Like this format, and be sure to see how the version number corresponds to the bottom of the this file.

> ## [8.0.2]
> ### Fixed
> - [#862](https://github.com/shakacode/react_on_rails/pull/862) by [justin808](https://github.com/justin808)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/877)*

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,10 @@ Changes since last non-beta release.
 
 *Please add entries here for your pull requests.*
 
+## [8.0.4]
```
Comment (justin808) [inline CHANGELOG.md:11]: see the bottom of the file for how you do the diff of 8.04 and 8.03, @chenqingspring

Comment (chenqingspring) [reply CHANGELOG.md:11]: ok. I didn't notice the references :(

but the build failed again..

**Comment by justin808** [comment]
Thanks @chenqingspring!

**Pull request merged**